Tests/IntegrationTests/SwiftPMTests.swift

```diff
@@ -201,4 +201,93 @@ private struct SwiftPMTests {
+            try localFileSystem.writeFileContents(
+                AbsolutePath(validating: "Tests/ReproTests/Subject.swift", relativeTo: packagePath),
+                string: """
+                struct Subject {
+                    static func a() { _ = "a" }
+                    static func b() { _ = "b" }
+                }
+                """
+            )
+            try localFileSystem.writeFileContents(
+                AbsolutePath(validating: "Tests/ReproTests/ReproTests.swift", relativeTo: packagePath),
+                string: """
+                import Testing
+                import class Foundation.ProcessInfo
+                @Suite struct Suite {
+                    @Test func testProfilePathCanary() throws {
+                        let pattern = try #require(ProcessInfo.processInfo.environment["LLVM_PROFILE_FILE"])
+                        #expect(pattern.hasSuffix(".%p.profraw"))
+                    }
+                    @Test func testA() async { await #expect(processExitsWith: .success) { Subject.a() } }
+                    @Test func testB() async { await #expect(processExitsWith: .success) { Subject.b() } }
+                }
+                """
+            )
+            let expectedCoveragePath = try await executeSwiftTest(packagePath, extraArgs: ["--show-coverage-path"]).stdout.trimmingCharacters(in: .whitespacesAndNewlines)
+            try await executeSwiftTest(packagePath, extraArgs: ["--enable-code-coverage", "--disable-xctest"])
+            let coveragePath = try AbsolutePath(validating: expectedCoveragePath)
+
+            // Check the coverage path exists.
+            #expect(localFileSystem.exists(coveragePath))
+
+            // This resulting coverage file should be merged JSON, with a schema that valiades against this subset.
+            struct Coverage: Codable {
+                var data: [Entry]
+                struct Entry: Codable {
+                    var files: [File]
+                    struct File: Codable {
+                        var filename: String
+                        var summary: Summary
+                        struct Summary: Codable {
+                            var functions: Functions
+                            struct Functions: Codable {
+                                var count, covered: Int
+                                var percent: Double
+                            }
+                        }
+                    }
+                }
+            }
+            let coverageJSON = try localFileSystem.readFileContents(coveragePath)
+            let coverage = try JSONDecoder().decode(Coverage.self, from: Data(coverageJSON.contents))
+
+            // Check for 100% coverage for Subject.swift, which should happen because the per-PID files got merged.
+            let subjectCoverage = coverage.data.first?.files.first(where: { $0.filename.hasSuffix("Subject.swift") })
+            #expect(subjectCoverage?.summary.functions.count == 2)
```

Comment:
thought: I'm not certain we should be validating the contents of the coverage data in this test as SwiftPM has no control over this.

Reply:
I actually think SwiftPM _does_ control this. It's SwiftPM that takes the LLVM raw profile data, in `.profraw` files, merges it into a `.profdata` file, and then creates a JSON file to represent the coverage.

The JSON file has a known schema which is API so we can rely on it and the test has been constructed above specifically so we can test that there's 100% coverage of a file, when split across two tests that run in separate processes.